pages/api_integration-full-node-streaming.md

```diff
@@ -0,0 +1,439 @@
+
+### StreamOrderbookFill/ClobMatch
+This message is only used to update fill amounts, it does not add or remove orders from the book but can change the quantity remaining for open orders.
+
+The `ClobMatch` data structure contains either a `MatchOrders` or a `MatchPerpetualLiquidation` object. Match Deleveraging events are not emitted. Within each Match object, a `MakerFill` array contains the various maker orders that matched with the singular taker order and the amount of quantums matched.
+
+Note that all matches occur at the maker order price. The `orders` field in the `StreamOrderbookFill` object allow for price lookups based on order id. It contains all the maker order ids, and in the case of non-liquidation orders, it has the taker order.
+
+Mapping each order in `orders` to the corresponding value in the `fill_amounts` field provides the absolute filled amount of quantums that each order is filled to after the ClobMatch was processed. 
+
+
+<details>
+
+<summary>Code Snippet</summary>
+
+```go
+// fillAmountMap is a map of order ids to fill amounts.
+// The SetOrderFillAmount code can be found in `the `OrderUpdateV1` section.
+func (c *GrpcClient) ProcessMatchOrders(
+	matchOrders *clobtypes.MatchOrders,
+	orderMap map[clobtypes.OrderId]clobtypes.Order,
+	fillAmountMap map[clobtypes.OrderId]uint64,
+) {
+	takerOrderId := matchOrders.TakerOrderId
+	clobPairId := takerOrderId.GetClobPairId()
+	localOrderbook := c.Orderbook[clobPairId]
+
+	indexerTakerOrder := v1.OrderIdToIndexerOrderId(takerOrderId)
+	localOrderbook.SetOrderFillAmount(&indexerTakerOrder, fillAmountMap[takerOrderId])
+
+	for _, fill := range matchOrders.Fills {
+		makerOrder := orderMap[fill.MakerOrderId]
+		indexerMakerOrder := v1.OrderIdToIndexerOrderId(makerOrder.OrderId)
+		localOrderbook.SetOrderFillAmount(&indexerMakerOrder, fillAmountMap[makerOrder.OrderId])
+	}
+}
+
+func (c *GrpcClient) ProcessMatchPerpetualLiquidation(
+	perpLiquidation *clobtypes.MatchPerpetualLiquidation,
+	orderMap map[clobtypes.OrderId]clobtypes.Order,
+	fillAmountMap map[clobtypes.OrderId]uint64,
+) {
+	localOrderbook := c.Orderbook[perpLiquidation.ClobPairId]
+	for _, fill := range perpLiquidation.GetFills() {
+		makerOrder := orderMap[fill.MakerOrderId]
+		indexerMakerOrderId := v1.OrderIdToIndexerOrderId(makerOrder.OrderId)
+		localOrderbook.SetOrderFillAmount(&indexerMakerOrderId, fillAmountMap[makerOrder.OrderId])
+	}
+}
+```
+
+</details>
+
+## FAQs
+
+### Optimistic Orderbook Execution
+
+By protocol design, each validator has their own version of the orderbook and optimistically processes orderbook matches. As a result, you may see interleaved sequences of order removals, placements, and state fill amount updates when optimistically processed orderbook matches are removed and later replayed on the local orderbook.
+
+![full node streaming diagram](../artifacts/full_node_streaming_diagram.jpg)
+
+Note that DeliverTx maps to exec mode `execModeFinalize`.
+
+### What Are the Exec Modes?
```

Comment:
To make this fit a little better on this page, I suggest either:
- renaming this to not be a question, e.g. "Exec Mode Reference", or
- moving it to the FAQ section and reformatting it there (Q: what are... )

I also think the code snippet could stay expanded because it helps break up the page at a glace